File: node/sdk/controllers/createStateController.js
import * as express from "express";
import { uuid } from "uuidv4";
import { dapr } from "../util/util.js";
const router = express.Router();

export const createStateController = router.post("/", async (req, res) => {
  try {
    const data = req.body.data;
    const orderId = data.orderId;

    const state = [
      {
        key: uuid(),
        value: data,
      },
    ];
    await dapr.client.state.save(dapr.stateStoreName, state);

    res.status(200).json({ msg: `Created order with Order ID: ${orderId}` });
  } catch (error) {
    console.log(error);
    res.status(500).json({ message: error });
  }
});

File: go/http/controllers/createState/createStateController.go
package createState

import (
	"bytes"
	"encoding/json"
	"fmt"
	"log"
	"net/http"
	"io"

	"github.com/google/uuid"
	constants "github.com/azureossd/dapr-state-management-examples/go/http/constants"
)

func CreateStateController(w http.ResponseWriter, r *http.Request) {
	uuid := uuid.New()
	requestBody, err := io.ReadAll(r.Body)
	if err != nil {
		log.Fatal(err)
	}
	// Give the "Key" a default value of a uuid
	state := []constants.State{
		{
			Key: uuid.String(),
		},
	}
	// Unmarshal the request body into the struct
	json.Unmarshal(requestBody, &state[0].Value)
	// Marshal the Struct Array into JSON to send in the POST request to Dapr
	t, err := json.Marshal(state)
	// If there is an error marhsalling state, return it
	if err != nil {
		log.Fatal(err)
	}
	res, err := http.Post("http://localhost:3500/v1.0/state/statestore", "application/json", bytes.NewBuffer(t))

	if err != nil {
		log.Fatal(err)
	}

	defer res.Body.Close()

	m := map[string]string{"msg": fmt.Sprintf("Order created with ID: %v", uuid.String())}
	e := json.NewEncoder(w)
	e.Encode(m)
}
